prediction_market_agent/agents/microchain_agent/app_nft_game.py

```diff
@@ -0,0 +1,169 @@
+"""
+PYTHONPATH=. streamlit run prediction_market_agent/agents/microchain_agent/app_nft_game.py
+
+Tip: if you specify PYTHONPATH=., streamlit will watch for the changes in all files, instead of just this one.
+"""
+
+import typing as t
+
+import streamlit as st
+from prediction_market_agent_tooling.tools.balances import get_balances
+from prediction_market_agent_tooling.tools.datetime_utc import DatetimeUTC
+
+from prediction_market_agent.agents.identifiers import AgentIdentifier
+from prediction_market_agent.agents.microchain_agent.agents_nft_game import (
+    AGENTS_IN_THE_GAME,
+    TREASURY_SAFE_ADDRESS,
+    NFTAgent,
+)
+from prediction_market_agent.agents.microchain_agent.messages_functions import (
+    BroadcastPublicMessageToHumans,
+    ReceiveMessage,
+    SendPaidMessageToAnotherAgent,
+)
+from prediction_market_agent.db.long_term_memory_table_handler import (
+    LongTermMemoryTableHandler,
+)
+from prediction_market_agent.db.prompt_table_handler import PromptTableHandler
+
+st.set_page_config(
+    page_title="Agent's NFT-locked Treasury Game", page_icon="🎮", layout="wide"
+)
+
+
+@st.cache_resource
+def long_term_memory_table_handler(
+    identifier: AgentIdentifier,
+) -> LongTermMemoryTableHandler:
+    return LongTermMemoryTableHandler.from_agent_identifier(identifier)
+
+
+@st.cache_resource
+def prompt_table_handler(identifier: AgentIdentifier) -> PromptTableHandler:
+    return PromptTableHandler.from_agent_identifier(identifier)
+
+
+def send_message_part(nft_agent: NFTAgent) -> None:
+    message = st.text_area("Write a message to the agent")
+
+    if st.button("Send message", disabled=not message):
+        # TODO: Don't do this manually with deployment private key, use the user's wallet!
+        SendPaidMessageToAnotherAgent()(nft_agent.wallet_address, message)
+        st.success("Message sent and will be processed soon!")
+
+
+def parse_function_and_body(
+    role: t.Literal["user", "assistant", "system"], message: str
+) -> t.Tuple[str | None, str | None]:
+    message = message.strip()
+
+    if role == "assistant":
```

Comment:
I am missing some comments here, an example would be very helpful.

Reply:
sure, added